plugin/src/py/android_screenshot_tests/recorder.py

```diff
@@ -88,12 +88,39 @@ def _clean(self):
             shutil.rmtree(self._output)
         os.makedirs(self._output)
 
+    def _draw_diff_outline(self, failure_file, diff, image_to_draw_on):
+        draw = ImageDraw.Draw(image_to_draw_on)
+        draw.rectangle(list(diff), outline=(255, 0, 0))
+        image_to_draw_on.save(failure_file)
+
     def _is_image_same(self, file1, file2, failure_file):
         with Image.open(file1) as im1, Image.open(file2) as im2:
             diff_image = ImageChops.difference(im1, im2)
```

Comment:
So it turned out that `ImageChops.difference` does not consider different image sizes as different images, but if only compares the intersections https://github.com/python-pillow/Pillow/blob/ee079ae67e7e24ec789d3cc7d180820a70d32fe6/src/libImaging/Chops.c#L74

This is definately not a bug in `facebook/screenshot-tests-for-android` but IMHO PIL can also claim that this is the intended behaviour. My suggestion is to mitigate this issue here, and check size differences after running `ImageChops.difference`